addons/stock/models/product.py

```diff
@@ -501,12 +501,14 @@ def write(self, vals):
             if done_moves:
                 raise UserError(_("You can not change the unit of measure of a product that has already been used in a done stock move. If you need to change the unit of measure, you may deactivate this product."))
         if 'type' in vals:
```

Comment:
It is problematic in the case some products were consumable and some stockable.  Just need if any('type' in vals and vals['type'] != x.type for x in self)